catalog/spatial/registry/registry-publication-action-provider/src/main/java/org/codice/ddf/registry/publication/manager/RegistryPublicationManager.java

```diff
@@ -0,0 +1,146 @@
+            return;
+        }
+
+        Attribute registryIdAttribute = mcard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID);
+        if (registryIdAttribute == null || StringUtils.isBlank(registryIdAttribute.getValue()
+                .toString())) {
+            return;
+        }
+        String registryId = registryIdAttribute.getValue()
+                .toString();
+
+        Attribute locationAttribute =
+                mcard.getAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS);
+        List<String> locations = new ArrayList<>();
+        if (locationAttribute != null) {
+            locations = locationAttribute.getValues()
+                    .stream()
+                    .map(Object::toString)
+                    .collect(Collectors.toList());
+        }
+        if (event.getTopic()
+                .equals(CREATED_TOPIC) || event.getTopic()
+                .equals(UPDATED_TOPIC)) {
+            publications.put(registryId, locations);
+        } else if (event.getTopic()
+                .equals(DELETED_TOPIC)) {
+            publications.remove(registryId);
+        }
+    }
+
+    public void init() throws FederationAdminException, PrivilegedActionException {
+
+        // Change from here
+        // Remove this subject stuff and change to use Security.runAsAdmin() once the PrivilegedException option has been implemented
+        Set<Principal> principals = new HashSet<>();
+        String localRoles = System.getProperty(KARAF_LOCAL_ROLES, "");
+        for (String role : localRoles.split(",")) {
+            principals.add(new RolePrincipal(role));
+        }
+        Subject subject = new Subject(true, principals, new HashSet(), new HashSet());
+
+        PrivilegedExceptionAction<List<Metacard>> federationAdminServiceAction =
+                () -> federationAdminService.getRegistryMetacards();
+        List<Metacard> metacards = subject.doAs(subject, federationAdminServiceAction);
+        // To here
+
+        for (Metacard metacard : metacards) {
+            Attribute registryIdAttribute =
+                    metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_ID);
+            if (registryIdAttribute == null || StringUtils.isBlank(registryIdAttribute.getValue()
+                    .toString())) {
+                LOGGER.warn("Warning metacard (id: {}} did not contain a registry id.",
+                        metacard.getId());
+                continue;
+            }
+            Attribute locations =
+                    metacard.getAttribute(RegistryObjectMetacardType.PUBLISHED_LOCATIONS);
+
+            String registryId = registryIdAttribute.getValue()
+                    .toString();
+            if (locations != null) {
+                publications.put(registryId,
+                        locations.getValues()
+                                .stream()
+                                .map(Object::toString)
+                                .collect(Collectors.toList()));
```

Comment:
do we want to make this list unmodifiable also otherwise when someone calls getPublications they can add to this list